scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/executors/InProcessTaskExecutor.java

```diff
@@ -200,15 +203,16 @@ public TaskResultImpl execute(TaskContext taskContext, PrintStream output, Print
      * @param scriptHandler
      * @param thirdPartyCredentials
      * @param variables             Environment variables.
+     * @param JobMap             Map of job results.
      * @return The result of the executed script.
      * @throws Throwable
      */
     private Serializable execute(TaskContext taskContext, PrintStream output, PrintStream error,
-            ScriptHandler scriptHandler, Map<String, String> thirdPartyCredentials, VariablesMap variables)
-            throws Throwable {
+            ScriptHandler scriptHandler, Map<String, String> thirdPartyCredentials, VariablesMap variables,
+            Map<String, Serializable> JobMap) throws Throwable {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Rename this local variable to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00117)